include/residuals.h

```diff
@@ -8,6 +8,13 @@
 
 namespace GLnexus {
 
+// Dataset and the records it has for a particular site
+struct DatasetSiteInfo {
+    std::string name;
+    std::shared_ptr<const bcf_hdr_t> header;
+    std::vector<std::shared_ptr<bcf1_t>> records;
+};
+
 class Residuals {
```

Comment:
Using a class saves passing some of the context arguments (contigs and samples), that is what this class is used for.